### PR TITLE
End to end tests.

### LINKED ISSUE #161: Trigger E2E tests after chart changes
Updates the CI/CD pipeline to run the E2E tests when the charts are changed. The E2E tests should validate if the chart does not have issues

### PR DESCRIPTION
**Pull request by jvanz**
## Description

Adds a new Github workflow to run the end to end tests on every PR into main branch. As well as changes in any branch and new tags.

Fix #161

### CODE REVIEW

**Review by viccuad — APPROVED** [review]
LGTM!

File: .github/workflows/e2e-tests.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      policy-server-tag: ${{ steps.get-policy-server.outputs.policy-server-tag }}
+    steps:
+      - name: "Checkout repository"
+        uses: actions/checkout@v3
+
+      - name: "Get controller container image"
+        id: get-controller
+        shell: bash
+        run: |
+          echo "controller-image-repository=$(helm show values charts/kubewarden-controller | yq  '.common.cattle.systemDefaultRegistry')/$(helm show values charts/kubewarden-controller | yq  '.image.repository')" >> $GITHUB_OUTPUT
```
Comment (kravciak) [inline .github/workflows/e2e-tests.yml:30]:
```suggestion
          echo "controller-image-repository=$(helm show values charts/kubewarden-controller | yq  '.common.cattle.systemDefaultRegistry+"/"+.image.repository')" >> $GITHUB_OUTPUT
```

File: .github/workflows/e2e-tests.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+          echo "controller-image-tag=$(helm show values charts/kubewarden-controller | yq  '.image.tag')" >> $GITHUB_OUTPUT
+
+      - name: "Get policy server container image"
+        id: get-policy-server
+        shell: bash
+        run: |
+          echo "policy-server-repository=$(helm show values charts/kubewarden-defaults | yq  '.common.cattle.systemDefaultRegistry')/$(helm show values charts/kubewarden-defaults | yq '.policyServer.image.repository')" >> $GITHUB_OUTPUT
```
Comment (kravciak) [inline .github/workflows/e2e-tests.yml:37]:
```suggestion
          echo "policy-server-repository=$(helm show values charts/kubewarden-defaults | yq  '.common.cattle.systemDefaultRegistry+"/"+.policyServer.image.repository')" >> $GITHUB_OUTPUT
```

**Review by kravciak — APPROVED** [review]
LGTM, just added two suggestions to make it more readable

**Pull request merged**